optuna/visualization/_parallel_coordinate.py

```diff
@@ -235,9 +230,9 @@ def _target(t: FrozenTrial) -> float:
                 ticktext=["{:.3g}".format(math.pow(10, x)) for x in tickvals],
             )
         elif _is_categorical(trials, p_name):
-            vocab: DefaultDict[Union[int, str], int] = defaultdict(lambda: len(vocab))
+            vocab: DefaultDict[int | str, int] = defaultdict(lambda: len(vocab))
```

Comment:
```suggestion
            vocab: defaultdict[int | str, int] = defaultdict(lambda: len(vocab))
```
It seems `defaultdict` is also available as a type annotation.
https://peps.python.org/pep-0585/#implementation

Reply:
Great suggestion. replaced.